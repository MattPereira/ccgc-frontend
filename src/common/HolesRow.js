import React from "react";

/**
 * Component for displaying hole numbers 1-18
 *
 *
 *
 */

const HolesRow = () => {
  return (
    <tr className="table-dark">
      <th>HOLE</th>
      {Array.from({ length: 18 }, (_, i) => (
        <th key={i + 1}>{i + 1}</th>
      ))}
      <th>TOT</th>
    </tr>
  );
};

export default HolesRow;
