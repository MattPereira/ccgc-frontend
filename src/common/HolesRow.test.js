import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import HolesRow from "./HolesRow";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <HolesRow updatePath="/courses/pebble-beach/edit" />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <HolesRow updatePath="/courses/pebble-beach/edit" />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("displays", function () {
  render(
    <MemoryRouter>
      <HolesRow />
    </MemoryRouter>
  );

  const row = screen.getByRole("row");
  expect(row).toBeInTheDocument();

  const holeTableHeader = screen.getByText("HOLE");
  expect(holeTableHeader).toBeInTheDocument();

  const totalTableHeader = screen.getByText("TOT");
  expect(totalTableHeader).toBeInTheDocument();
});
