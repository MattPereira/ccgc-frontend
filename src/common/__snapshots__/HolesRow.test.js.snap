// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  HOLE123456789101112131415161718TOT
</DocumentFragment>
`;
